components/NavBar.vue

```diff
@@ -7,32 +7,80 @@
       </div>
     </div>
     <div class="nav-right">
-      <input type="text" />
+      <Transition>
+        <input
+          type="text"
+          
+          v-model="searchKey"
+          v-if="showSearchField"
+        />
+      </Transition>
       <!-- for search -->
-      <span class="search-icon">
+      <span class="search-icon" @click="onSearchIconClick">
         <img :src="searchIcon" />
       </span>
       <select @change="switchLanguage">
-        <option  :selected="item.code === $i18n.defaultLocale" v-for="item in $i18n.locales" :key="item.code" :value="item.code">{{ item.name }}</option>
+        <option
+          :selected="item.code === $i18n.defaultLocale"
+          v-for="item in $i18n.locales"
+          :key="item.code"
+          :value="item.code"
+        >
+          {{ item.name }}
+        </option>
       </select>
-    </div> 
+    </div>
   </nav>
 </template>
 
 <script>
+import debounce from "@/helpers/debounce";
+import { mapState } from "vuex";
 export default {
   data() {
     return {
+      showSearchField: false,
       logo: require("@/assets/img/leaf.svg"),
       searchIcon: require("@/assets/img/search.svg"),
     };
   },
+  computed: {
+    ...mapState("todo", ["searchText"]),
+    searchKey: {
+      get() {
+        return this.searchText
+      },
+      set(newValue) {
+        this.debouncedSearch(newValue)
+      }
+    }
```

Comment:
Can we use searchKey as normal data instead of computed property? Because we should only set computed properties in very specific cases
More [here](https://vuejs.org/guide/essentials/computed.html#avoid-mutating-computed-value)

Reply:
searchKey is dependent on state variable searchText. If I reset searchText the search input field should clear accordingly. thats why using computed, so that the change is triggered automatically.

Reply:
What's the difference between `searchKey` and `searchText`?

Reply:
No difference. I didn't want to assign a state variable directly as v-model. please see [here](https://stackoverflow.com/questions/54928111/should-we-use-v-model-to-modify-vuex-store#:~:text=Yes%20you%20can%20but%20is,the%20control%20over%20the%20state.)

Reply:
So rather than setting a computed property, we can use the searchKey from store as getter and use mutations to set the value as use changes it. What do you think?

Reply:
So what do you suggest I should put inside the v-model?

Reply:
We can debounce `@input` and make API call / get search data and set the `searchKey` that is in the store using mutation, while we can get the same value from store as a getter and use it as `value` property of input box.

Reply:
Got it